### PR TITLE
Strong Typing

### PR DESCRIPTION
**Pull request by gronke**
Another attempt to enforce typed interfaces without getting eaten by merge conflicts.

### CODE REVIEW

File: .travis/mypy-stubs/libzfs.pyi [hunk 1]
```diff
@@ -336,53 +336,3 @@ class ZfsConverter:
     __init__ = ... # type: Any
     to_native = ... # type: Any
     to_property = ... # type: Any
-
-class datetime(date):
```
Comment (igalic) [inline .travis/mypy-stubs/libzfs.pyi]: oh?

Comment (gronke) [reply .travis/mypy-stubs/libzfs.pyi]: It was never used in our context, but caused confusion. That's why I have removed it. Do you think that was an error?

File: iocage/lib/Releases.py [hunk 2]
```diff
@@ -31,19 +31,23 @@
 
 # MyPy
 import libzfs
+ReleaseListType = typing.List[iocage.lib.Release.ReleaseGenerator]
```
Comment (igalic) [inline iocage/lib/Releases.py]: more of this please!

File: iocage/lib/Config/Jail/BaseConfig.py [hunk 3]
```diff
@@ -146,11 +146,22 @@ def update_special_property(self, name: str) -> bool:
...
+        )
+    ) -> None:
         """Attach a special property to the configuration."""
         self.special_properties[name] = special_property
 
-    def _set_legacy(self, value, **kwargs) -> None:
+    def _set_legacy(  # noqa: T484
+        self,
+        value: typing.Union[bool, str],
+        **kwargs
```
Comment (igalic) [inline iocage/lib/Config/Jail/BaseConfig.py]: please open up an issue for cleaning up `noqa: T484`

Comment (gronke) [reply iocage/lib/Config/Jail/BaseConfig.py]: @igalic as you wish #282

**Pull request merged**